sale_order_line_date/models/sale_order.py

```diff
@@ -0,0 +1,30 @@
+# © 2016 OdooMRP team
+# © 2016 AvanzOSC
+# © 2016 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# © 2016 Eficent Business and IT Consulting Services, S.L.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    @api.onchange('commitment_date')
+    def _onchange_commitment_date(self):
+        """Warn if the commitment dates is sooner than the commitment date"""
+        result = super(SaleOrder, self)._onchange_commitment_date()
```

Comment:
All super could be simplified (not blocking) :

```suggestion
        result = super()._onchange_commitment_date()
```

Reply:
I don't think it is a good way to do date.
I personally prefer being explicit, considering the result is the same.
Do as you want